tests/apollo/test_skvbc_chaotic_startup.py

```diff
@@ -237,24 +218,114 @@ async def test_f_staggered_replicas_requesting_vc(self, bft_network, skvbc, nurs
 
         await trio.sleep(5)  # TODO: remove when bft_network.wait_for_view also waits for system liveness.
 
-        for r in bft_network.all_replicas(without={replica_to_stop}):
+        await self._verify_replicas_are_in_view(view, bft_network.all_replicas(without={replica_to_stop}), bft_network)
+
+        await self._verify_the_system_is_able_to_make_progress(bft_network, skvbc, expected_next_primary, bft_network.all_replicas(without={replica_to_stop}))
+
+    @with_trio
+    @with_bft_network(start_replica_cmd_with_vc_timeout("20000"),
+                      selected_configs=lambda n, f, c: f >= 2)
+    @with_constant_load
+    async def test_f_minus_one_staggered_replicas_requesting_vc(self, bft_network, skvbc, nursery):
+        """
+        In this test we check that if f-1 replicas have started to run and initiated viewchange, then the system is
+        still able to make progress. To make sure the system still have 2f+1 active replicas, we wait for the early
+        replicas to complete their state transfer before stopping the primary.
+
+        For that we perform the following steps:
+        1. start f - 1 replicas
+        2. wait for those replicas to initiate view change
+        3. start the rest 2f + 2 replicas
+        4. make sure the system is able to make progress
+        5. stop the current primary
+        6. wait for the system to complete a view change
+        7. make sure the system is able to make progress
+        """
+        n = bft_network.config.n
+        f = bft_network.config.f
+        c = bft_network.config.c
+
+        initial_view = initial_primary = 0
+        expected_next_view = expected_next_primary = 1
+
+        # We start by choosing a random set of f - 1 replicas.
+        excluded = {initial_primary}
+        early_replicas = bft_network.random_set_of_replicas(f - 1, without=excluded)
+
+        print(f"STATUS: Starting F={f - 1} replicas.")
+        bft_network.start_replicas(replicas=early_replicas)
+        await self._wait_for_less_than_f_plus_one_replicas_to_initiate_viewchange(early_replicas, bft_network)
+        print("STATUS: Early replicas started and initiated View Change.")
+
+
+        late_replicas = bft_network.all_replicas(without=early_replicas)
+        print(f"STATUS: Starting the remaining {n - f + 1} replicas.")
+        bft_network.start_replicas(late_replicas)
+
+        view = await bft_network.wait_for_view(
+            replica_id=initial_primary,
+            expected=lambda v: v == initial_primary,
```

Comment:
For clarity might be would be better to replace "initial_primary" with "initial_view".

Reply:
done